core/vm/contracts_stateful.go

```diff
@@ -171,6 +173,67 @@ func (c *nativeAssetCall) Run(evm *EVM, caller ContractRef, addr common.Address,
 	return ret, remainingGas, err
 }
 
+// exportAsset atomically transfers a native asset to a recipient address as well as calling that
+// address
+type exportAsset struct {
+	gasCost uint64
+}
+
+func PackExportAssetInput(address common.Address, assetID common.Hash, assetAmount *big.Int, callData []byte) []byte {
+	input := make([]byte, 84+len(callData))
+	copy(input[0:20], address.Bytes())
+	copy(input[20:52], assetID.Bytes())
+	assetAmount.FillBytes(input[52:84])
+	copy(input[84:], callData)
+	return input
+}
+
+func UnpackExportAssetInput(input []byte) (string, string, *big.Int, error) {
+	if len(input) < 109 {
+		return "", "", nil, fmt.Errorf("export asset call input had unexpcted length %d", len(input))
+	}
+
+	recipient := string(input[:45]) // string encoded X-* address.
+
+	// TODO: confirm an encoded assetID is 32 bytes in length
+	assetID := string(common.TrimLeftZeroes(input[45:77])) // string encoded assetID. could be 0 left padded literal "AVAX".
+
+	assetAmount := new(big.Int).SetBytes(input[77:109])
+
+	return recipient, assetID, assetAmount, nil
+}
+
+// Run implements StatefulPrecompiledContract
+func (c *exportAsset) Run(evm *EVM, caller ContractRef, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	// input: encodePacked(address 20 bytes, assetID 32 bytes, assetAmount 32 bytes, callData variable length bytes)
+	if suppliedGas < c.gasCost {
+		return nil, 0, ErrOutOfGas
+	}
+	remainingGas = suppliedGas - c.gasCost
+
+	if readOnly {
+		return nil, remainingGas, ErrExecutionReverted
+	}
+
+	recipient, assetID, assetAmount, err := UnpackExportAssetInput(input)
+	if err != nil {
+		return nil, remainingGas, ErrExecutionReverted
+	}
+
+	// create an export transaction to export the asset to the recipient address
+	var keys []*avaCrypto.PrivateKeySECP256K1R
```

Comment:
The `newExportTx` call in `vm` requires a slice of keys in order to construct and sign the transaction. I wasn't able to find anything like that at this point, so I'm guessing this is the incorrect approach. Feedback here would be much appreciated.